src/MicroFocus.FAS.Adapters.FileSystem/FileSystemAdapter.cs

```diff
@@ -0,0 +1,140 @@
+                }
+
+                await handler.QueueItemAsync(repositoryItem.ItemId,
+                                             new FileContents(() =>
+                                                              {
+                                                                  var file = new FileInfo(repositoryItem.Metadata.ItemLocation);
+                                                                  return file.OpenRead();
+                                                              }),
+                                             repositoryItem.Metadata);
+            }
+        }
+
+        public ChangeType CompareItems(IRepositoryItem? existingItem, IRepositoryItem currentItem)
+        {
+            throw new NotImplementedException();
+        }
+
+        private async Task ProcessDirectoryAsync(DirectoryInfo directory, IFileListResultsHandler handler, CancellationToken cancellationToken)
+        {
+            try
+            {
+                foreach (var file in directory.EnumerateFiles("*", SearchOption.TopDirectoryOnly))
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    await handler.QueueItemAsync(new ItemMetadata(file.Name, file.FullName)
+                                                 {
+                                                     Size = file.Length,
+                                                     ModifiedTime = file.LastWriteTimeUtc,
+                                                     AccessedTime = file.LastAccessTimeUtc,
+                                                     CreatedTime = file.CreationTimeUtc
+                                                 },
+                                                 file.DirectoryName ?? string.Empty,
+                                                 cancellationToken);
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Failed to process directory {Directory}", directory);
+                await handler.RegisterFailureAsync(directory.FullName, new FailureDetails($"Failed to process directory {directory}", ex));
+            }
+
+            try
+            {
+                var directories = directory.EnumerateDirectories();
+                foreach (var subDirectory in directories)
+                {
+                    await ProcessDirectoryAsync(subDirectory, handler, cancellationToken);
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Failed to process subdirectories for {Directory}", directory);
+                await handler.RegisterFailureAsync(directory.FullName, new FailureDetails($"Failed to process directory {directory}", ex));
+            }
+        }
+
+        private void LogIn(string userName, string password)
```

Comment:
This was added for QA to validate that encrypted properties are handled correctly. It only logs the values. That's also why the username and password is in the required properties. It also makes sense for file system if we want to impersonate a user when accessing a shared folder.
Even though now it's not doing anything, I'm inclined to keep it, or at least the required properties. We will add impersonation to this adapter at some point and this will avoid changing the adapter configuration.